### PR TITLE
[5421] When duplicating items or content that open 'groovy' or 'ftl' editors and then closing the editor, UI should call the item_unlock API

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/5421

### CODE REVIEW

File: ui/app/src/state/epics/content.ts [hunk 1]
```diff
@@ -283,12 +286,14 @@ const content: CrafterCMSEpic[] = [
...
             const editableAsset = isEditableAsset(payload.path);
             if (editableAsset) {
+              const extension = getFileExtensionFromPath(path);
               return showCodeEditorDialog({
                 authoringBase: state.env.authoringBase,
                 site: state.sites.active,
                 path,
-                type: 'asset',
-                onSuccess: payload.onSuccess
+                mode: getEditorModeFromExtension(extension),
```
Comment (jvega190) [inline ui/app/src/state/epics/content.ts:294]: @rart this is not directly related with the original issue, I need to discuss this changes with you

File: ui/app/src/state/epics/content.ts [hunk 2]
```diff
@@ -281,14 +282,16 @@ const content: CrafterCMSEpic[] = [
...
+            const mode = getEditorMode(state.content.itemsByPath[payload.path].mimeType);
             const editableAsset = isEditableAsset(payload.path);
             if (editableAsset) {
               return showCodeEditorDialog({
                 authoringBase: state.env.authoringBase,
                 site: state.sites.active,
                 path,
-                type: 'asset',
-                onSuccess: payload.onSuccess
+                mode,
```
Comment (jvega190) [inline ui/app/src/state/epics/content.ts:292]: `type` was being sent instead of `mode`, so the editor mode wasn't properly set

**Review by rart — APPROVED** [review]
